redash/query_runner/treasuredata.py

```diff
@@ -71,23 +76,24 @@ def type(cls):
     def __init__(self, configuration):
         super(TreasureData, self).__init__(configuration)
 
-    def get_schema(self):
+    def get_schema(self, get_stats=False):
         schema = {}
-        try:
-            with tdclient.Client(self.configuration.get('apikey')) as client:
-                for table in client.tables(self.configuration.get('db')):
-                    table_name = '{}.{}'.format(self.configuration.get('db'), table.name)
-                    for table_schema in table.schema:
-                        schema[table_name] = {'name': table_name, 'columns': table.schema}
-        except Exception, ex:
-            raise Exception("Failed getting schema")
+        if self.configuration.get('getschema') == True:
```

Comment:
The `default` property in the schema is just a hint for the UI, it doesn't set actual values. So in your case change this line to: `if self.configuration.get('getschema', True):` to have `True` as default value.